### DISCUSSION TITLE
verifytypes + `@contextmanager` complains about default param and missing docstring, despite `--ignoreexternal`

### DISCUSSION THREAD

**Discussion by jakkdl** [microsoft/pyright#8905] [Q&A] [answered]
I'm not sure this is a bug in pyright, or something that would need to be addressed in typeshed - but even if it's the latter I feel like running with `--ignoreexternal` should be silencing them.

```python
from contextlib import contextmanager

@contextmanager
def fail_at() -> Generator[int, None, None]:
    "blabla"
    ...
```
```sh
$ pyright --pythonversion=3.12 --verifytypes=foo --ignoreexternal
[...]
Symbols without documentation:
  Functions without docstring: 1
  Functions without default param: 1
  Classes without docstring: 0
[...]
$ pyright --pythonversion=3.12 --verifytypes=foo
[...]
  Functions without docstring: 1
  Functions without default param: 1
  Classes without docstring: 3
[...]
```

**Comment by erictraut** [comment]
I'm not able to repro the problem as described. Here's what I get when I create a fake "py.typed" package with a single `__init__.py` file that has the above code pasted into it.

```
$ pyright --pythonversion=3.12 --verifytypes test_pkg1 --ignoreexternal
[...]
Symbols without documentation:
  Functions without docstring: 0
  Functions without default param: 0
  Classes without docstring: 0
[...]
$ pyright --pythonversion=3.12 --verifytypes test_pkg1
[...]
Symbols without documentation:
  Functions without docstring: 0
  Functions without default param: 0
  Classes without docstring: 3
[...]
```

Please double check that you're able to repro the output you posted above.

You might also find the `--verbose` output useful in helping to diagnose the problem.

**Reply by jakkdl** [reply]
argh, this is making me insane. I'm now failing to re-repro it, but still getting it in a different venv and it is happening in CI (https://github.com/python-trio/trio/actions/runs/10722754974/job/29734470633?pr=3010#step:5:1366)

I'll see if I manage to get the repro back or debug it somehow, or I'll maybe just ignore it in CI for now

**Reply by erictraut** [reply]
OK, I'm happy to investigate further if you find some additional clues.

**Reply by jakkdl** [reply]
okay I'm able to reliably repro it, though currently far from minimal. This bash script will exhibit it for me:
```sh
cd /tmp
mkdir new_repro
cd new_repro
git clone git@github.com:jakkdl/trio.git
cd trio
git checkout repro_pyright_verifytypes_fail_at
python -m venv .venv
source .venv/bin/activate
pip install . pyright
pyright --verifytypes=trio --ignoreexternal --verbose | grep fail_at
```

```
[...]
   trio.fail_at
trio.fail_at
  /tmp/new_repro/trio/.venv/lib/python3.12/site-packages/trio/_timeouts.py:99:16 - warning: One or more default values in function "trio.fail_at" is specified as "..."
```

**Reply by erictraut** [reply, marked as answer]
Thanks for putting together the repro. I was able to diagnose the cause of the issue and fix it. This will be addressed in the next release.

**Reply by jakkdl** [reply]
Awesome, thank you!
